contracts/vault/AssetVault.sol

```diff
@@ -215,6 +217,27 @@ contract AssetVault is IAssetVault, OwnableERC721, Initializable, ERC1155Holder,
         emit Call(msg.sender, to, data);
     }
 
+    /**
+     * @notice Approve a token for spending by an external contract. Note that any token
+     *         approved in the whitelist does not make good collateral, because the allowed
+     *         spender may be able to withdraw it from the vault.
+     *
+     * @param token                 The token to approve.
+     * @param spender               The approved spender.
+     * @param amount                The amount to approve.
+     */
+    function callApprove(address token, address spender, uint256 amount) external override onlyWithdrawDisabled nonReentrant {
+        if (msg.sender != owner() && !ICallDelegator(owner()).canCallOn(msg.sender, address(this)))
+            revert AV_CallDisallowed(msg.sender);
+
+        if (!CallWhitelistApprovals(address(whitelist)).isApproved(token, spender)) revert AV_NonWhitelistedApproval(token, spender);
+
+        // Do approval
+        IERC20(token).approve(spender, amount);
+
+        emit Approve(msg.sender, token, spender);
```

Comment:
Could adding the `amount` that is approved in the `Approve` event be userful to the front end ?

Reply:
Yes, good call - will add